LEGO1/mxtransitionmanager.cpp

```diff
@@ -38,12 +41,110 @@ MxResult MxTransitionManager::Tickle()
   return 0;
 }
 
+// OFFSET: LEGO1 0x1004bc30 STUB
+void MxTransitionManager::EndTransition(MxBool)
+{
+  // TODO
+}
+
+// OFFSET: LEGO1 0x1004bd10
+void MxTransitionManager::Transition_Dissolve()
+{
+  // If the animation is finished
+  if (m_animationTimer == 40) {
+    m_animationTimer = 0;
+    EndTransition(TRUE);
+    return;
+  }
+
+  // If we are starting the animation
+  if (m_animationTimer == 0) {
+    // Generate the list of columns in order...
+    for (int i = 0; i < 640; i++) {
+      m_columnOrder[i] = i;
+    }
+
+    // ...then shuffle the list (to ensure that we hit each column once)
+    for (i = 0; i < 640; i++) {
+      int swap = rand() % 640;
```

Comment:
We should probably use `MxS32` for all the loop bars in here

Reply:
For `swap`? Or loop counters `i` and `j`?

Reply:
For all of the variables I'd say - or is there a good reason not to?

Reply:
No reason other than the habit of using int for the loop counter. I changed the types over.